src/DotNet.Testcontainers/Containers/ResourceReaper.cs

```diff
@@ -147,23 +178,37 @@ private async Task StartAsync(CancellationToken ct = default)
       await this.resourceReaperContainer.StartAsync(ct)
         .ConfigureAwait(false);
 
-      _ = this.MaintainRyukConnection(ryukInitializedTaskSource, ct);
+      this.connectionTask = this.MaintainRyukConnection(ryukInitializedTaskSource);
 
-      await ryukInitializedTaskSource.Task
-        .ConfigureAwait(false);
+      await Task.WhenAny(
+        Task.Delay(Timeout.InfiniteTimeSpan, ct),
+        ryukInitializedTaskSource.Task
+      ).ConfigureAwait(false);
```

Comment:
I am not sure if there is a more elegant way to wait for the `ryukInitializedTaskSource.Task` either to complete or for `ct` to be cancelled.

Reply:
This shouldn't be different to:

```cs
await ryukInitializedTaskSource.Task
  .ConfigureAwait(false);
```

There must be a different error. Which test faild?

Reply:
In the current code the CancellationToken `ct` would otherwise not cancel the `StartAsync` method. Instead the method would block until the initialization was successful or the `InitTimout` was elapsed.

I think the test `GetAndStartNewAsyncShouldBeACancellableDuringConnectionInitialization` would fail if only `await ryukInitializedTaskSource.Task.ConfigureAwait(false);` was used.

Reply:
That is a good idea. But `ct` should cancel `MaintainRyukConnection` only during connection initialization.

Reply:
Sorry, I don't get it. Probably I need to take a closer look at it again. As far as I understand it, `ct` can not cancel the connection initialization right now. It gets not passed to `MaintainRyukConnection`.

But it controls the termination of `StartAsync` instead. If someone cancels `ct` from outside, `Task.Delay` will finish. This can be every time. Before, while and after the connection initialization.

`MaintainRyukConnection` knows nothing about the cancellation, the task **continues to run**.

Sorry for the back and forth, but some parts are just too complex 😂.

Reply:
Sorry for the confusion. I have reverted this part again and pass `ct` (now called specifically `initCancellationToken`) to `MaintainRyukConnection` again. I have also moved the the timeout mechanism out of the `MaintainRyukConnection` method into `GetAndStartNewAsync`, so that `MaintainRyukConnection` has only to care about one `initCancellationToken` and one general connection maintenance token.

The `initCancellationToken` is ignored as soon as the connection initialization was successful.

Reply:
👍 I can take a look at it on the weekend.